compiler/noirc_evaluator/src/acir/mod.rs

```diff
@@ -1071,11 +1071,16 @@ impl<'a> Context<'a> {
         result: AcirVar,
         bit_size: u32,
         binary: &Binary,
-        dfg: &DataFlowGraph,
         predicate: AcirVar,
     ) -> Result<(), RuntimeError> {
-        let Some(msg) = binary.check_unsigned_overflow_msg(dfg, bit_size) else {
-            return Ok(());
+        let msg = match binary.operator {
+            BinaryOp::Add { unchecked: false } => "attempt to add with overflow",
+            BinaryOp::Sub { unchecked: false } => "attempt to subtract with overflow",
+            BinaryOp::Mul { unchecked: false } if bit_size != 1 => {
```

Comment:
Why are we not handling this case within SSA?

Reply:
I explained it a bit in "Additional Context" though I might have wrote too much 😅 

I pushed a commit for this. I was mainly trying to avoid producing a larger diff, but changing this makes sense.